test/unit/test_inventory.py

```diff
@@ -96,5 +96,53 @@ def test_roles_in_inventory(self):
         self.assertIn('master', roles)
         self.assertIn('control-plane', roles)
 
+    def test_internal_address_inventory(self):
+        inventory = demo.generate_inventory()
+        for node in inventory['nodes']:
+            node.pop('address')
+
+        cluster = demo.new_cluster(inventory)
+        for node in cluster.inventory['nodes']:
+            self.assertNotIn('address', node)
+
+        final_inventory = utils.get_final_inventory(cluster, inventory)
+        for node in final_inventory['nodes']:
+            self.assertNotIn('address', node)
+
+    def test_internal_address_remove_node_inventory(self):
+        inventory = demo.generate_inventory(**demo.MINIHA_KEEPALIVED)
+        for node in inventory['nodes']:
+            node.pop('address')
+        procedure_inventory = {
+            'nodes': [copy.deepcopy(inventory['nodes'][0])]
+        }
+
+        # Remove node inventory
+        context = demo.create_silent_context(procedure='remove_node')
+        cluster = demo.new_cluster(inventory, procedure_inventory=procedure_inventory, context=context)
+        for node in cluster.inventory['nodes']:
+            self.assertNotIn('address', node)
+
+        final_inventory = utils.get_final_inventory(cluster)
```

Comment:
It is preferable to pass original inventory to the utils.get_final_inventory(cluster, inventory), because it is always passed that one during real Kubemarine work.

Reply:
Hmm, actually there are two cases: 1) generation of ansible inventory, passed without second argument 2) recreation of the inventory, passed with the second argument. Ok then, let's leave it this way.